python/tk_multi_publish2/dialog.py

```diff
@@ -1478,6 +1484,38 @@ def _show_no_items_error(self):
         self.ui.main_stack.setCurrentIndex(self.PUBLISH_SCREEN)
         self._overlay.show_no_items_error()
 
+    def _validate_task_required(self):
+        """
+        Validates that a task is selected for every item and disables/enables the
+        validate and publish buttons and finally change the color for the task
+        label.
+        """
+        # Avoid validation if the setting `task_required` is False or not exists
+        if not self._bundle.get_setting("task_required"):
```

Comment:
we should declare the `task_required` setting to the `info.yml` manifest as well.